src/tools/yarprobotstatepublisher/src/robotstatepublisher.cpp

```diff
@@ -217,12 +221,26 @@ void YARPRobotStatePublisherModule::onRead(yarp::rosmsg::sensor_msgs::JointState
         return;
     }
 
-    // Check if joint states contain at least as many joints of the model
-    if (v.name.size() < m_jointPos.size())
-    {
-        yError() << "Size mismatch. Model has " << m_jointPos.size()
-                 << " joints, while the received JointState message has " << v.name.size() << " joints.";
-        return;
+    if (m_model_joints == "lower") {
+
+        // Check if the joints states are more than the number of joints of the model
+        if (v.name.size() > m_jointPos.size()) {
+            yError() << "Size mismatch. Model has " << m_jointPos.size()
+                     << " joints, while the received JointState message has " << v.name.size() << " joints.";
+            return;
+        }
```

Comment:
If this is the case, in this case, what is the value contained in `m_jointPos` for joints that are not contained in the joint state topic? To avoid confusion, I think we should document it.

Reply:
@traversaro at the moment I am not certain how and when this case occurred but @lrapetti and I ran into this problem a while back with another robot model we were using for retargeting. Unfortunately, we cannot recall the exact steps to reproduce the problem.